handler/src/main/java/io/netty/handler/ssl/TlsMasterKeyHandler.java

```diff
@@ -0,0 +1,166 @@
+     * @param sessionId The sessionId An arbitrary byte sequence chosen by the server to identify an
+     *                  active or resumable session state.
+     */
+    protected abstract void accept(SecretKey masterKey, byte[] sessionId);
+
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
+        //only try to log the session info if the ssl handshake has successfully completed.
+        if (evt == SslHandshakeCompletionEvent.SUCCESS) {
+
+            boolean shouldHandle = SystemPropertyUtil.getBoolean(SYSTEM_PROP_KEY, false);
+
+            //if the handler is not an instance of DoNothingTlsMasterKeyBiConsumer try to handle it
+            if (shouldHandle) {
+                final SslHandler handler = ctx.pipeline().get(SslHandler.class);
+                final SSLEngine engine = handler.engine();
+                final SSLSession sslSession = engine.getSession();
+
+                //the OracleJVM does not expose a way to get the master secret, so try to use reflection to get it.
+                try {
+                    if (isSunSslEngineAvailable() && sslSession.getClass().equals(SSL_SESSIONIMPL_CLASS)) {
+                        SecretKey secretKey = (SecretKey) SSL_SESSIONIMPL_MASTER_SECRET_FIELD.get(sslSession);
+                        accept(secretKey, sslSession.getId());
+                    }
+                    //TODO: handle OpenSSL as well
+                } catch (IllegalAccessException e) {
+                    throw new IllegalArgumentException("Failed to access the field 'masterSecret' via reflection.", e);
+                }
+            }
+        }
+
+        ctx.fireUserEventTriggered(evt);
+    }
+
+    /**
+     * Record the session identifier and master key to the {@link InternalLogger} named <code>io.netty.wireshark</code>.
+     * ex. <code>RSA Session-ID:XXX Master-Key:YYY</code>
+     * This format is understood by Wireshark 1.6.0.
+     * https://code.wireshark.org/review/gitweb?p=wireshark.git;a=commit;h=686d4cabb41185591c361f9ec6b709034317144b
+     * The key and session identifier are forwarded to the log named 'io.netty.wireshark'.
+     */
+    public static class WiresharkTlsMasterKeyHandler extends TlsMasterKeyHandler {
+
+        private static final InternalLogger wireshark_logger =
+                InternalLoggerFactory.getInstance("io.netty.wireshark");
+
+        @Override
+        protected void accept(SecretKey masterKey, byte[] sessionId) {
+            if (masterKey.getEncoded().length != 48) {
+                throw new IllegalArgumentException("An invalid length master key was provided.");
+            }
+            wireshark_logger.warn("RSA Session-ID:{} Master-Key:{}",
+                    javax.xml.bind.DatatypeConverter.printHexBinary(sessionId).toLowerCase(),
+                    javax.xml.bind.DatatypeConverter.printHexBinary(masterKey.getEncoded()).toLowerCase());
```

Comment:
can we get rid of `javax.xml.bind.DatatypeConverter` ? This is not accessible in Java9+ easily